Assets/MRTK/SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -208,6 +209,24 @@ private Vector3 WorldSpacePressDirection
             }
         }
 
+        /// <summary>
+        /// The press direction of the button as defined by a NearInteractionTouchableSurface, in local space,
+        /// using Vector3.forward as an optional fallback when no NearInteractionTouchableSurface is defined.
+        /// </summary>
+        private Vector3 LocalSpacePressDirection
+        {
+            get
+            {
+                var nearInteractionTouchable = GetComponent<NearInteractionTouchableSurface>();
```

Comment:
Since it looks like we call `GetComponent<NearInteractionTouchableSurface>();` both here and in `WorldSpacePressDirection`, every time they're called, it's probably worth caching the result and only trying to get the component again if the cached value turns null.

Reply:
@keveleigh this was something that I was going to change in a later PR. There are pretty egregious examples of non-cached references _all_ over the code, but I was going to put them in separately (discussed this with @thalbern)

Reply:
@Zee2 Perfect, thanks!